src/genai/credentials.py

```diff
@@ -19,3 +22,11 @@ def __init__(
         if api_endpoint is None:
             raise ValueError("api_endpoint must be provided")
         self.api_endpoint = api_endpoint
+        self.remove_version()
+
+    def remove_version(self) -> None:
+        parsed_url = parse_url(self.api_endpoint)
+
+        if parsed_url.path and "/v" in parsed_url.path:
```

Comment:
This is an incorrect assumption. Some environments can be hidden behind some path. 
Just check if the path ends with `vX` (X = `numeric version`)

`re.match(r".\d$", api_endpoint)`